MoviesApp/Model/MoviesListProvider.swift

```diff
@@ -50,26 +50,40 @@ class MoviesListManager {
         NotificationCenter.default.post(name: .ImageLoaded, object: movie)
     }
     
+    func setDetails(for movie: Movie, details: Details){
+        let index = getIndexOfSortedMovie(movie)
+        allMovies[index].details = details
+    }
+    
+    func setBackDrop(for movie: Movie, image: UIImage) {
+        let index = getIndexOfSortedMovie(movie)
+        allMovies[index].setBackdropImage(image)
+    }
+    
     private func allImagesSet() -> Bool {
         return allMovies.filter { $0.posterImage != nil }.count == allMovies.count
     }
     
-    func getIndexOfSortedMovie(_ movie: Movie) -> Int{
+    func getIndexOfSortedMovie(_ movie: Movie) -> Int {
```

Comment:
I'm not sure that using the title to identify a movie is a good idea... maybe you have more than one movie with the same title (different years, durations, etc).. can you think of something better to use here?